test/integration/spec/remotetracks.js

```diff
@@ -34,6 +35,234 @@ function getTracksOfKind(participant, kind) {
+        assert.equal(aliceRemoteAudioTrack.switchOffReason, null);
+      });
+    });
+
+    after(() => {
+      [aliceRoom, bobRoom, charlieRoom].forEach(room => room && room.disconnect());
+    });
+  });
+
+  // TODO(mmalavalli) Enable test once SFU implements this behavior.
+  describe.skip('enable/disable', () => {
+    let aliceRoom;
+    let bobRoom;
+    let charlieRoom;
+    let daveRoom;
+
+    before(async () => {
+      [, aliceRoom, [bobRoom, charlieRoom]] = await waitFor(setup({
+        testOptions: { audio: true, video: false },
+        otherOptions: { audio: true, video: false },
+        participantNames: ['Alice', 'Bob', 'Charlie'],
+        nTracks: 1
+      }), 'Alice, Bob and Charlie to join a Room');
+
+      daveRoom = await waitFor(connect(getToken('Dave'), Object.assign({
+        audio: false,
+        name: aliceRoom.name,
+        video: false
+      }, defaults)), 'Dave to join a Room');
+
+      await waitFor(
+        participantsConnected(daveRoom, 3),
+        'Alice, Bob and Charlie to be visible as RemoteParticipants'
+      );
+
+      const participants = [...daveRoom.participants.values()];
+
+      await waitFor(
+        participants.map(participant => tracksSubscribed(participant, 1)),
+        'Dave to subscribe to Alice, Bob and Charlie\'s AudioTracks'
+      );
+
+      const daveRemoteTracks = participants
+        .flatMap(participant => [...participant.audioTracks.values()])
+        .map(({ track }) => track);
+
+      await waitFor(
+        new Promise(resolve => {
+          let nSwitchedOn = 0;
+          daveRemoteTracks.map(trackSwitchedOn).forEach(switchedOn => switchedOn.then(() => {
+            nSwitchedOn++;
+            // TODO(mmalavalli): Update max tracks switched on limit once SFU sets the final number.
+            if (nSwitchedOn >= 2) {
+              resolve();
+            }
+          }));
+        }),
+        'Max AudioTracks for Dave to be switched on'
+      );
+    });
+
+    ['Alice', 'Bob', 'Charlie'].forEach(identity => {
+      [false, true].forEach(isEnabled => {
+        context(`when ${identity} ${isEnabled ? 'enables' : 'disables'} the LocalAudioTrack`, () => {
+          let remoteTrack;
+
+          before(() => {
+            const room = [aliceRoom, bobRoom, charlieRoom].find(({ localParticipant }) => localParticipant.identity === identity);
+            const localTrack = room.localParticipant.audioTracks.values().next().value.track;
+            localTrack[isEnabled ? 'enable' : 'disable']();
+            const { track } = daveRoom.participants.get(room.localParticipant.sid).audioTracks.values().next().value;
+            remoteTrack = track;
+            return waitFor([
+              { true: trackSwitchedOn, false: trackSwitchedOff }[isEnabled](remoteTrack),
+              trackEnabled(remoteTrack, isEnabled)
+            ], `${identity}'s RemoteAudioTrack to be ${isEnabled ? 'switched on and enabled' : 'switched off and disabled'}`);
+          });
```

Comment:
I found this expression very complicated to read, how about making this function `async` and:
```suggestion
           await  trackEnabled(remoteTrack, isEnabled);
           await isEnabled ? waitFor(trackSwitchedOn(remoteTrack), 'track switched on') :  waitFor(trackSwitchedOff(remoteTrack), 'track switched off')
```